include/boost/url/query_param.hpp

```diff
@@ -115,16 +117,49 @@ struct query_param
 
     /** Constructor
     */
+    query_param() = default;
+
+    /** Constructor
+    */
+    template<class Allocator>
     query_param(
-        query_param_view const& v)
-        : key(v.key)
+        query_param_view const& v,
+        Allocator const& a)
+        : key(v.key, a)
         , value(v.has_value ?
             v.value :
-            string_view())
+            string_view(), a)
         , has_value(v.has_value)
     {
     }
 
+    /** Constructor
+    */
+    template <class Allocator = std::allocator<char>>
+    query_param(
+        const_string const& key_,
+        const_string const& value_,
+        bool has_value_)
+        : key(key_)
+        , value(value_)
+        , has_value(has_value_)
+    {
+    }
+
+    /** Constructor
+    */
+    template <class Allocator = std::allocator<char>>
```

Comment:
I think this overload should accept `const_string::factory const&` instead of `Allocator const&`, we don't need a separate _Allocator_ overload. That's actually a pessimization since we will be constructing two different factories for the same allocator (one for the key and one for the value).